pipeline/terraform/modules/es_index/main.tf

```diff
@@ -0,0 +1,26 @@
+locals {
+  full_name = var.name
+
+  mappings_file = "${var.config_path}/mappings.${var.mappings_name}.json"
+  analysis_name = var.analysis_name != "" ? var.analysis_name : var.mappings_name
+
+  analysis_json = jsondecode(local.analysis_name != "empty" ? file("${var.config_path}/analysis.${local.analysis_name}.json") : "{}")
+
+  analysis = {
+    analyzer : try(jsonencode(local.analysis_json["analyzer"]), "{}")
```

Comment:
TIL: Terraform has a `try` function!

Do we need `try(…)` here? Is there any case when we wouldn't want this to fail if passed invalid JSON?

Reply:
The problem I was overcoming here is that I don't want to have to specify all the analysis properties in the JSON even if we don't want them.

I think this is the clearest way to express "get the property or return an empty object".

The JSON is known to be well-formed, having been decoded into an object above, so this is just about whether the requested property exists at the top level.  It's possible that the property could be present but inappropriate (e.g. a number or a string), but that will be caught by `elasticstack_elasticsearch_index`, and wouldn't be caught here even if we didn't `try`.

TF is a bit annoying about optional or default properties and `try` expects both results to be the same type, meaning that `{}` is not the same as `{"x":{"y":"z"}}` because it doesn't have an `x` property, hence encoding it to a string.   (it may be possible to do something by defining a `map` of `maps` and telling it how to decode the JSON in the first place, instead of letting it do type magic on its own, but since we just want a string representation, that seemed like a bit of a faff)